drivers/include/periph/i2c.h

```diff
@@ -140,147 +145,110 @@ int i2c_init_master(i2c_t dev, i2c_speed_t speed);
  * @param[in] dev       I2C device to access
  *
```

Comment:
as the bus is shared, we have the situation that multiple devices use the bus with different clock speeds. By moving the clock selection into the acquire function, this is now possible.